README.md

```diff
@@ -1,88 +1,158 @@
+```
+
+Once the client has been instantiated, it can be used to issue queries. For example, to create
+an instance in an existing class named `test` with the data: `{ testField: 'testValue' }`:
 
+```javascript
+var createP = client.query(q.Create(Ref('classes/test'), { testField: 'testValue' }));
+```
 
-## Building it yourself
+Note that we are using the `Ref` alias defined above here.
 
-### Setup
+All methods on `faunadb.Client` return [ES6 Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+So, if we wanted to transform the Promise to return the `Ref` of the newly created instance:
 
-    npm install
+```javascript
+createP.then(function(res) {
+  return res.ref;
+});
+```
 
+`res` is a JSON object containing the FaunaDB response. See the JSDocs for `faunadb.Client`,
+and the [FaunaDB Developer Guide](https://faunadb.com/documentation/dev) for more information on responses.
 
-### Build
+#### Pagination Helpers
 
-`npm run build` or `npm run watch`.
+This driver contains helpers to provide a simpler API for consuming paged
+responses from FaunaDB. See the [FaunaDB Developer Guide](https://faunadb.com/documentation/dev) and
+the [Paginate Function Reference](https://faunadb.com/documentation/queries#read_functions-paginate_set)
+for a description of paged responses.
 
+Using the helper to page over sets lets the driver handle cursoring and pagination state. For example:
 
-### Test
+```javascript
+var helper = client.paginate(q.Match(Ref('indexes/test_index'), 'example-term'));
+```
 
-    npm run test
+Here, `helper` is an instance of `PageHelper`. The methods `eachPage` and `eachItem`
+will execute a callback function on each consumed page, or each item in each page, respectively.
 
-To run tests, you will need to create a `testConfig.json` in the faunadb-js directory
-that looks like this:
+```javascript
+helper.eachPage(function(page) {
+  console.log(page); // Will log the page's contents, for example: [ Ref("classes/test/1234"), ... ]
+});
+```
 
-    {
-      "domain": "localhost",
-      "scheme": "http",
-      "port": 8443,
-      "auth": "secret"
-    }
+Note that `eachPage` and `eachItem` return a `Promise<void>` that is fulfilled on the completion
+of pagination.
 
-`domain`, `scheme`, and `port` are optional and will default to FaunaDB cloud.
-So to test with cloud, `testConfig.json` is as simple as:
+The pagination can be transformed server-side via the FaunaDB query language.
+For example, to retrieve the matched instances:
 
-    {
-      "auth": "your_fauna_key"
-    }
+```javascript
+helper.map(function(ref) { return q.Get(ref); }).eachPage(function(page) {
+  console.log(page); // Will now log the retrieved instances.
+});
+```
 
-You can also set the `FAUNA_DOMAIN`, `FAUNA_SCHEME`, `FAUNA_PORT`, and `FAUNA_ROOT_KEY` environment variables.
+[See the JSDocs](https://faunadb.github.com/faunadb-js/PageHelper.html) for more information on the pagination helper.
```

Comment:
I think it's missing a clear definition on which functions have their effects on the server (`map`, `filter`, e.g) and which ones are local (`eachPage`, `nextPage`, `previousPage`, e.g).